src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/NamingStyles/Serialization/SymbolSpecification.cs

```diff
@@ -363,10 +363,25 @@ public SymbolKindOrTypeKind(MethodKind methodKind) : this()
 
             public bool MatchesSymbol(ISymbol symbol)
                 => SymbolKind.HasValue ? symbol.IsKind(SymbolKind.Value) :
-                   TypeKind.HasValue ? symbol is ITypeSymbol type && type.TypeKind == TypeKind.Value :
+                   TypeKind.HasValue ? symbol is ITypeSymbol type && MatchesTypeSymbolKind(type.TypeKind, TypeKind.Value) :
                    MethodKind.HasValue ? symbol is IMethodSymbol method && method.MethodKind == MethodKind.Value :
                    throw ExceptionUtilities.Unreachable;
 
+            private static bool MatchesTypeSymbolKind(TypeKind kind1, TypeKind kind2)
+            {
+                if (kind1 == CodeAnalysis.TypeKind.Module && kind2 == CodeAnalysis.TypeKind.Class)
+                {
+                    return true;
+                }
+
+                if (kind1 == CodeAnalysis.TypeKind.Class && kind2 == CodeAnalysis.TypeKind.Module)
+                {
+                    return true;
+                }
+
+                return kind1 == kind2;
+            }
```

Comment:
This is a bad change, I think. It resulted from the tests creating `SymbolSpecification`s themselves, and not relying on the parser. Either tests should use the parser, or the following is updated to include modules:

https://github.com/dotnet/roslyn/blob/3e091aba2b46f8574c41f8431dcbeba42274ca86/src/EditorFeatures/DiagnosticsTestUtilities/NamingStyles/NamingStylesTestOptionSets.cs#L129-L136